syntax = "proto2";

import "nanopb.proto";

message greet_packet {
  optional string destination_name = 1 [(nanopb).max_size = 255];
  optional uint32 source_port = 2;
}

message client_list_request {

}

message c2d_envelope {
  oneof payload {
    greet_packet greet = 1;
    client_list_request client_list_request = 2;
  }
}
